### PR TITLE
chore: add ESRP CDN for browser downloads

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: packages/playwright-core/src/server/registry/browserFetcher.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export async function downloadBrowserWithProgressBar(title: string, browserDirec
 
   const zipPath = path.join(os.tmpdir(), downloadFileName);
   try {
-    const retryCount = 3;
+    const retryCount = 5;
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/registry/browserFetcher.ts:39]: Here we either need to decide between:

a) Continuing trying only 3 times, so the other 2 mirrors are basically unused
b) Retrying up to 5 times so we keep using the other 2 mirrors

did b) for now.

File: packages/playwright-core/src/server/registry/index.ts [hunk 2]
```diff
@@ -37,6 +37,10 @@ const PACKAGE_PATH = path.join(__dirname, '..', '..', '..');
 const BIN_PATH = path.join(__dirname, '..', '..', '..', 'bin');
 
 const PLAYWRIGHT_CDN_MIRRORS = [
+  'https://playwright.azureedge.net/dbazure/download/playwright', // ESRP
+  'https://playwright.download.prss.microsoft.com/dbazure/download/playwright', // ESRP Fallback
+
+  // Old CDNs, to be removed.
   'https://playwright.azureedge.net',
   'https://playwright-akamai.azureedge.net',
```
Comment (yury-s) [inline packages/playwright-core/src/server/registry/index.ts:45]: Can you also add a comment here that akamai and verizon are actually pointing to a different falvor of azure cdn and will soon become same?

File: packages/playwright-core/src/server/registry/browserFetcher.ts (hunk 1 above)
Comment (yury-s) [reply packages/playwright-core/src/server/registry/browserFetcher.ts:39]: 2) would be staying on the safe side and I vote for it.

File: packages/playwright-core/src/server/registry/browserFetcher.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ export async function downloadBrowserWithProgressBar(title: string, browserDirec
...
     for (let attempt = 1; attempt <= retryCount; ++attempt) {
```
Comment (yury-s) [inline packages/playwright-core/src/server/registry/browserFetcher.ts:40]: Replave `retryCount` with `downloadURLs.length`, otherwise it's hard to follow where 5 comes from, given that the list of CDN urls is in a different file.

**Review by yury-s — APPROVED** [review]

File: packages/playwright-core/src/server/registry/browserFetcher.ts (hunk 3 above)
Comment (mxschmitt) [reply packages/playwright-core/src/server/registry/browserFetcher.ts:40]: Soon we have downloadURLs = 2 and we want to keep retryCount of 3 so having them aligned seem to be misleading.

**Review by Skn0tt — APPROVED** [review]
looks good!
